src/main/java/net/neoforged/neoforge/network/PacketDistributor.java

```diff
@@ -5,259 +5,129 @@
-        }
-    }
-
-    private final BiFunction<PacketDistributor<T>, T, Consumer<Packet<?>>> functor;
-    private final PacketFlow flow;
-
-    public PacketDistributor(BiFunction<PacketDistributor<T>, T, Consumer<Packet<?>>> functor, PacketFlow flow) {
-        this.functor = functor;
-        this.flow = flow;
-    }
-
-    public PacketDistributor(Function<PacketDistributor<T>, Consumer<Packet<?>>> functor, PacketFlow flow) {
-        this((d, t) -> functor.apply(d), flow);
+    public static void sendToPlayer(ServerPlayer player, CustomPacketPayload... payloads) {
+        player.connection.send(makeClientboundPacket(payloads));
     }
 
     /**
-     * Apply the supplied value to the specific distributor to generate an instance for sending packets to.
-     * 
-     * @param input The input to apply
-     * @return A curried instance
+     * Send the given payload(s) to all players in the given dimension
      */
-    public PacketTarget with(T input) {
-        return new PacketTarget(functor.apply(this, input), this);
+    public static void sendToPlayersInDimension(ServerLevel level, CustomPacketPayload... payloads) {
+        level.getServer().getPlayerList().broadcastAll(makeClientboundPacket(payloads), level.dimension());
     }
 
     /**
-     * Apply a no argument value to a distributor to generate an instance for sending packets to.
-     *
-     * @see #ALL
-     * @see #SERVER
-     * @return A curried instance
+     * Send the given payload(s) to all players in the area covered by the given radius around the given coordinates
+     * in the given dimension, except the given excluded player if present
      */
-    public PacketTarget noArg() {
-        return new PacketTarget(functor.apply(this, null), this);
-    }
-
-    private Consumer<Packet<?>> playerConsumer(final ServerPlayer entityPlayerMP) {
-        return p -> entityPlayerMP.connection.send(p);
-    }
-
-    private Consumer<Packet<?>> playerListDimConsumer(final ResourceKey<Level> dimensionType) {
-        return p -> getServer().getPlayerList().broadcastAll(p, dimensionType);
-    }
-
-    private Consumer<Packet<?>> playerListAll() {
-        return p -> getServer().getPlayerList().broadcastAll(p);
+    public static void sendToPlayersNear(
+            ServerLevel level,
+            @Nullable ServerPlayer excluded,
+            double x,
+            double y,
+            double z,
+            double radius,
+            CustomPacketPayload... payloads) {
+        Packet<?> packet = makeClientboundPacket(payloads);
+        level.getServer().getPlayerList().broadcast(excluded, x, y, z, radius, level.dimension(), packet);
     }
 
-    private Consumer<Packet<?>> clientToServer() {
-        return p -> Objects.requireNonNull(Minecraft.getInstance().getConnection()).send(p);
+    /**
+     * Send the given payload(s) to all players on the server
+     */
+    public static void sendToAllPlayers(CustomPacketPayload... payloads) {
```

Comment:
We might want to ask for an explicit `MinecraftServer` parameter here. Or maybe not.

Reply:
Hmm, not sure, to be honest. On one hand, it would be nice for type safety, but on the other hand, not having to provide the context feels better API-wise 🤔